client/landing/stepper/declarative-flow/internals/steps-repository/importer/index.tsx

```diff
@@ -87,6 +83,10 @@ export function withImporterWrapper( Importer: ImporterCompType ) {
 		const fromSiteData = useSelector( getUrlData );
 		const stepNavigator = useStepNavigator( flow, navigation, siteId, siteSlug, fromSite );
 		const currentPath = window.location.pathname + window.location.search;
+		const isRequestingAllSites = useSelector( isRequestingSites );
```

Comment:
Switching to the `isRequestingSites` selector means that the downstream loading state is taking a lot longer for me, because we're waiting for all the sites to be re-fetched before we proceed. When using the `hasAllSitesList` selector, we're only waiting for the sites to be populated for the first time, not necessarily from the fetch triggered by this page.

To fix this initial issue, I think it may be worth keeping the original logic that checks `hasAllSitesList`, but add in the fetch for the current site. I have created https://github.com/Automattic/wp-calypso/issues/102281 to track the work to fully switch the component tree over to request only the current site.

Reply:
Updated here [ba35b15](https://github.com/Automattic/wp-calypso/pull/102277/commits/ba35b158c7b0e8e30e20050829e03477247d39fe)